### PR TITLE
DOC: constant check in series

### LINKED ISSUE #54033: ENH: series.is_constant
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

Pandas Series have a property [`is_unique`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.is_unique.html) that returns true iff all values are distinct. 

In practice, one often needs to know if a Series is constant. Getting a functional result when series do not contain nans is straight forward: `(series.values[0] == series.values).all()`. However, [in the wild](https://github.com/search?q=%22nunique%28%29+%3D%3D+1%22+language%3APython+&type=code) we observe that users often simply write `series.nunique() == 1`. This approach is more costly for larger series, as it first computes the number of unique values before comparing.

My assumption is that introducing a dedicated `is_constant` property will help users choose the more performant option. Another assumption is that its acceptable that all-nan series are somewhat slower.

perf_short_circuit

```python
def setup(n):
    return pd.Series(list(range(n)))
```

perf_worst

```python
def setup(n):
    return pd.Series([1] * (n - 1) + [2])
```

### Feature Description

```python
@property
def is_constant():
    if v.shape[0] == 0:
        return False
    return (v[0] == v).all()
```

(based on `is_unique` and `nunique`, in the absence of nans)

To extend to NA values:
If `dropna=True`: add `v = remove_na_arraylike(v)`
If  `dropna=False`: add `or not pd.notna(v).any()` (note: differs from `np.unique` for `pd.NA` and `np.nan` mixed series by lack of disambiguation between the two)

### Alternative Solutions

Recommend the approach above for `series.nunique() == 1` patterns in the documentation ([here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.nunique.html))

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by sbrugman**
- [X] closes #54033
- [X] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [X] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [X] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [X] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/user_guide/cookbook.rst [hunk 1]
```diff
@@ -1488,3 +1488,34 @@ of the data values:
...
+
+   v = s.values
+   is_constant = v.shape[0] > 0 and (s[0] == s).all()
+
+This approach assumes that the series does not contain missing values.
+For the case that we would drop NA values, we can simply remove those values first:
+
+.. ipython:: python
+
+   from pandas.core.dtypes.missing import remove_na_arraylike
```
[the remark this answers is not in the document]
Comment (sbrugman) [reply doc/source/user_guide/cookbook.rst]: Replaced with `dropna`

File: doc/source/user_guide/cookbook.rst [hunk 2]
```diff
@@ -1488,3 +1488,31 @@ of the data values:
...
+Constant series
+---------------
+
+To assess if a series has a constant value, we can check if ``series.nunique() == 1``.
+However, a more performant approach, that does not count all unique values first, is:
+
+.. ipython:: python
+
+   v = s.values
+   is_constant = v.shape[0] > 0 and (s[0] == s).all()
```
Comment (rhshadrach) [inline doc/source/user_guide/cookbook.rst:1501]: I would describe the empty Series as being constant; much in the same way an empty function is: https://math.stackexchange.com/questions/1949584/is-the-empty-function-constant

Comment (sbrugman) [reply doc/source/user_guide/cookbook.rst:1501]: I agree (and it means that the commonly used `nunique() == 1` should be `nunique() <= 1`). 
Let me update the snippet

File: doc/source/user_guide/cookbook.rst [hunk 3]
```diff
@@ -1488,3 +1488,31 @@ of the data values:
...
+
+Constant series
+---------------
+
+To assess if a series has a constant value, we can check if ``series.nunique() <= 1``.
+However, a more performant approach, that does not count all unique values first, is:
+
+.. ipython:: python
+
+   v = s.values
```
Comment (mroeschke) [inline doc/source/user_guide/cookbook.rst:1500]: Could you use `.to_numpy()` instead of `.values` in these examples?

Comment (sbrugman) [reply doc/source/user_guide/cookbook.rst]: Done

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @sbrugman

**Pull request merged**